### DISCUSSION TITLE
Disabling Chrome save address popups during a headless run

### DISCUSSION THREAD

**Discussion by firepro20** [cypress-io/cypress#28701] [Questions and Help] [answered]
I was wondering if it is possible to do this as I searched for quite a while. I did come across a plugin however it does not work as intended. During my signup/registration test, Chrome sends out a save address pop up which I would like to silence or disable.

**Comment by portyu9** [comment, marked as answer]
Hi @firepro20 

This is now handled by Cypress itself.

Starting with **Cypress 15.6.0**, Cypress disables Chrome's address and credit-card autofill popups during test execution. So the preferred solution today is simply to upgrade Cypress rather than maintain a browser-launch workaround.

If you are pinned to an older Cypress release, the equivalent workaround is:

```js
const { defineConfig } = require('cypress')

module.exports = defineConfig({
  e2e: {
    setupNodeEvents(on, config) {
      on('before:browser:launch', (browser, launchOptions) => {
        if (browser.family === 'chromium' && browser.name !== 'electron') {
          launchOptions.preferences.default.autofill ??= {}
          launchOptions.preferences.default.autofill.profile_enabled = false
        }

        return launchOptions
      })
    },
  },
})
```

But on Cypress 15.6.0+ that preference is already applied by Cypress, so I would remove the custom workaround after upgrading.

Please mark this answer as accepted if it helped, thank you.
